packages/cfa-template/template/frontend/src/components/Integration/ModalFormBasedAuth.js

```diff
@@ -2,184 +2,207 @@ import Form from '@rjsf/core';
+    return (
+      <label htmlFor={id} className="block text-sm mb-4">
+        <span className="text-gray-700">
+          {label} {required ? '*' : null}
+        </span>
+        {description}
+        {children}
+        <p className="inline-flex pt-2 text-xs font-medium text-red-300">
+          {errors}
+        </p>
+        <p className="inline-flex pt-2 text-xs font-medium text-red-300">
+          {help}
+        </p>
+      </label>
+    );
+  }
+
+  function InputTextWidget(props) {
+    return (
+      <input
+        type="text"
+        className="block w-full mt-1 text-sm form-input rounded-lg"
+        value={props.value}
+        required={props.required}
+        placeholder={props.placeholder}
+      />
+    );
+  }
+
+  function InputPasswordWidget(props) {
+    return (
+      <input
+        type="password"
+        className="block w-full mt-1 text-sm form-input rounded-lg"
+        value={props.value}
+        required={props.required}
+        placeholder={props.placeholder}
+      />
+    );
+  }
+
+  function InputCheckboxWidget(props) {
+    return (
+      <>
+        <label htmlFor="custom-checkbox">{props.label}</label>
+        <Toggle
+          id="custom-checkbox"
+          className={props.value ? 'checked' : 'unchecked'}
+          onClick={() => props.onChange(!props.value)}
+          defaultChecked={props.value}
+        />
+      </>
+    );
+  }
+
+  const widgets = {
+    text: InputTextWidget,
+    password: InputPasswordWidget,
+    CheckboxWidget: InputCheckboxWidget,
+  };
+
+  return (
+    <>
+      <div className="fixed inset-0 z-30 flex items-end bg-black bg-opacity-50 sm:items-center sm:justify-center">
+        <div
+          className="w-full px-6 py-4 overflow-hidden bg-white rounded-t-lg sm:rounded-lg sm:m-4 sm:max-w-xl"
+          role="dialog"
+          id="modal"
+        >
+          <div className="mt-4 mb-6">
+            <p className="text-lg font-semibold text-gray-700 mb-6">
+              Authorize {name}
+            </p>
+
+            {isLoading ? (
+              <div className="flex justify-center px-8 py-8">
+                <svg
+                  className="animate-spin h-10 w-10 text-purple-400"
+                  fill="none"
+                  viewBox="0 0 24 24"
+                >
+                  <circle
+                    className="opacity-25"
+                    cx="12"
+                    cy="12"
+                    r="10"
+                    stroke="currentColor"
+                    strokeWidth="4"
+                  ></circle>
+                  <path
+                    className="opacity-75"
+                    fill="currentColor"
+                    d="M4 12a8 8 0 018-8V0C5.373 0 0 5.373 0 12h4zm2 5.291A7.962 7.962 0 014 12H0c0 3.042 1.135 5.824 3 7.938l3-2.647z"
+                  ></path>
+                </svg>
+              </div>
+            ) : null}
+
+            <Form
```

Comment:
Thought we had swapped out already (for JSONforms). Different place?

Reply:
Yes, we have a JSONForms version of this, I'll remove this one.